File: server/src/main/java/com/group2/server/model/AdoptionRequest.java
package com.group2.server.model;

import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDateTime;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public class AdoptionRequest {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private Long dogId;
    private Long userId;
    private LocalDateTime requestDate;
    private String status;
    private String userEmail;
}

File: client/app/src/main/java/com/group2/client/model/AdoptionRequest.kt
//AdoptionRequest model
package com.group2.client.model

import com.google.gson.annotations.SerializedName
import java.time.LocalDateTime

data class AdoptionRequest(
    @SerializedName("id") val id: Int? = null, // ID can be null when creating a new request, Change to Int
    @SerializedName("dogId") val dogId: Int, // Change to Int
    @SerializedName("userId") val userId: Int, // Change to Int
    @SerializedName("requestDate") val requestDate: String? = null,
    @SerializedName("status") val status: String? = null,
    @SerializedName("userEmail") val userEmail: String,
)